repositories/case_repository.go

```diff
@@ -504,6 +504,8 @@ func (repo *MarbleDbRepository) GetNextCase(ctx context.Context, exec Executor,
 				"assigned_to": nil,
 			},
 			squirrel.NotEq{"status": "closed"},
+			squirrel.NotEq{"id": c.Id},
+			squirrel.GtOrEq{"created_at": c.CreatedAt},
```

Comment:
As outlined verbally, this would fail if we are already on the most recent unassigned case, we basically reversed the initial issue.